### ISSUE TITLE
Tag: Inner span value not rendered

### ISSUE THREAD

**Issue by ozziej** [primefaces/primefaces#8724]
**Describe the defect**
when adding a p:tag component it does NOT add the value attribute

**Reproducer**

**Environment:**
 - PF Version: _11.0.3_
 - PF Theme: All
 - JSF + version: Any
 - Affected browsers: _e.g. All

**To Reproduce**
Add a p:tag for example:

<p:tag styleClass="p-mr-2" value="Primary"></p:tag>
or
<p:tag styleClass="p-mr-2" value="Primary"/>

**Expected behavior**
This should render in the browser as:

```html
<span id="j_idt311" class="ui-tag ui-widget p-mr-2"><span class="ui-tag-value">Primary</span></span>
```
Instead it renders as :
```html
<span id="j_idt311" class="ui-tag ui-widget p-mr-2"></span>
```
Its missing the inner <span> attribute
The reason for this is that the code was recently changed with fix : https://github.com/primefaces/primefaces/issues/8526

Specifically:

```java
String value = ComponentUtils.getValueToRender(context, tag);
```
This variable now returns null.

The method ComponentUtils.getValueToRender checks if the variable tag is an instance of **ValueHolder**, which it is NOT.
This method then simply returns null.

This is the problem.

**Example XHTML**
```html
<h:form id="frmTest">
   <p:tag styleClass="p-mr-2" value="Primary"></p:tag>
</h:form>
```

**Example Bean**
N/A

**Comment by melloware** [comment]
Hmmm but the only code changed was 1 line...

**Was:**
`writer.write(value);`

**Now:**
`writer.writeText(value, "value");`

The line you are referring to...

```java
String value = ComponentUtils.getValueToRender(context, tag);
```
Has been there since #8147 

See commit: https://github.com/primefaces/primefaces/pull/8147/files

**Comment by melloware** [comment]
cc @NicolaIsotta who made the original fix.

**Comment by melloware** [comment]
Ahh this must be an Elite issue as in 11.0.0 it works correctly: https://www.primefaces.org/showcase/ui/misc/tag.xhtml

So this must have broken when they patched 11.0.3 I will mark it Elite.

**Comment by ozziej** [comment]
Thanks. I did specifically mention it was 11.0.3.
